boa/src/syntax/lexer/tests.rs

```diff
@@ -443,55 +541,88 @@ fn test_regex_literal_flags() {
     let mut lexer = Lexer::new("1+ 1");
     lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
     assert_eq!(lexer.tokens[1].kind, TokenKind::Punctuator(Punctuator::Add));
-    assert_eq!(lexer.tokens[2].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[2].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
 }
 
 #[test]
 fn test_addition_no_spaces_right_side() {
     let mut lexer = Lexer::new("1 +1");
     lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
     assert_eq!(lexer.tokens[1].kind, TokenKind::Punctuator(Punctuator::Add));
-    assert_eq!(lexer.tokens[2].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[2].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
 }
 
 #[test]
 fn test_addition_no_spaces_e_number_left_side() {
     let mut lexer = Lexer::new("1e2+ 1");
     lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::NumericLiteral(100.0));
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Racional(100.0))
+    );
     assert_eq!(lexer.tokens[1].kind, TokenKind::Punctuator(Punctuator::Add));
-    assert_eq!(lexer.tokens[2].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[2].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
 }
 
 #[test]
 fn test_addition_no_spaces_e_number_right_side() {
     let mut lexer = Lexer::new("1 +1e3");
     lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::NumericLiteral(1.0));
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Integer(1))
+    );
     assert_eq!(lexer.tokens[1].kind, TokenKind::Punctuator(Punctuator::Add));
-    assert_eq!(lexer.tokens[2].kind, TokenKind::NumericLiteral(1000.0));
+    assert_eq!(
+        lexer.tokens[2].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Racional(1000.0))
+    );
 }
 
 #[test]
 fn test_addition_no_spaces_e_number() {
     let mut lexer = Lexer::new("1e3+1e11");
     lexer.lex().expect("failed to lex");
-    assert_eq!(lexer.tokens[0].kind, TokenKind::NumericLiteral(1000.0));
+    assert_eq!(
+        lexer.tokens[0].kind,
+        TokenKind::NumericLiteral(NumericLiteral::Racional(1000.0))
+    );
     assert_eq!(lexer.tokens[1].kind, TokenKind::Punctuator(Punctuator::Add));
     assert_eq!(
         lexer.tokens[2].kind,
-        TokenKind::NumericLiteral(100_000_000_000.0)
+        TokenKind::NumericLiteral(NumericLiteral::Racional(100_000_000_000.0))
```

Comment:
If we start seeing this a bit too wordy, we could implement `From<f64>` and `From<i32>` for `NumericLiteral`, and then add a `fn numeric_literal<L>(lit: L) -> Self where L: Into<NumericLiteral> {}` to `TokenKind`, which would directly allow us to pass a `f64` or a `i32`. I think I already did something similar in the modular parser PR.